### ISSUE TITLE
Uploading files to S3 over SSL is far slower than aws-cli

### ISSUE THREAD

**Issue by aaaristo** [aws/aws-sdk-js#596]
I'm uploading a 653MB file using this code, from an EC2 instance (c3.4xlarge, us-east-1):

```javascript
var AWS= require('aws-sdk'),
    fs= require('fs');

var params = {Bucket: 'bk-name', Key: 'andrea/test.json', Body: fs.createReadStream('../tmp/s3/test.json') };

var upload = new AWS.S3.ManagedUpload({ queueSize: 10, params: params});
 upload.send(function(err, data) {
    console.log(err, data);
 })
```
and is taking ~12secs.

The same code over HTTP takes ~7secs
```javascript
var AWS= require('aws-sdk'),
    fs= require('fs');

var params = {Bucket: 'bk-name', Key: 'andrea/test.json', Body: fs.createReadStream('../tmp/s3/json.json') };

var upload = new AWS.S3.ManagedUpload({ queueSize: 10, params: params, service: new AWS.S3({ sslEnabled: false }) });
 upload.send(function(err, data) {
    console.log(err, data);
 })
```

while aws-cli is taking ~4secs
```bash
aws s3 cp ../tmp/s3/test.json s3://bk-name/andrea/test.json
```

Any idea?

**Comment by lsegal** [comment]
@aaaristo have you tried playing with the queueSize? A c3.4xlarge would likely have more I/O and CPU resources to handle potentially more concurrent requests. It's also possible that you are saturating the I/O or CPU with 10 and it needs to be reduced. Playing with the value should provide some different performance characteristics.

You may also want to make sure you're using Node.js 0.12.x, which supports request keep-alive, something 0.10.x does not, and may affect performance.

**Comment by AdityaManohar** [comment]
@aaaristo 
I've tried to replicate this issue with a 1 GB file on an OS X laptop with 8GB of memory on Node.js v0.10.38 and SDK version 2.1.36. I am not seeing any difference in the time taken upload this file to Amazon S3. Here are the results:

```sh
$>node s3_perf.js
Upload time: 93.46s
```
```sh
$>./s3_perf.sh
Upload time: 95s
```
In fact, in my tests, the JavaScript SDK is consistently about 1 second faster.

There were some performance improvements and bug fixes pertaining to `AWS.S3.upload()` that landed in v2.1.28 of the SDK. I would recommend upgrading the SDK to v2.1.28 or higher and checking if you can still replicate the issue.

**Comment by AdityaManohar** [comment]
@aaaristo I'm marking this as closed as I'm not able to replicate any difference in upload times between the AWS CLI and the JavaScript SDK. Like I mentioned earlier, I would recommend upgrading to `v2.1.28` of the SDK or higher to see if you can still reproduce the issue. Feel free to reopen this issue or open another issue if you are still seeing a difference in upload times between the AWS CLI and the JavaScript SDK.
